Wire-iOS/Sources/UserInterface/Calling/BottomSheetContainerViewController/CallingActionsView/CallingActionsView.swift

```diff
@@ -0,0 +1,253 @@
+    private var videoButtonDisabledTapRecognizer: UITapGestureRecognizer?
+    private var verticalStackViewTopContraint: NSLayoutConstraint!
+
+    // Buttons
+    private let microphoneButton = CallingActionButton.microphoneButton()
+    private let cameraButton = CallingActionButton.cameraButton()
+    private let speakerButton = CallingActionButton.speakerButton()
+    private let flipCameraButton = CallingActionButton.flipCameraButton()
+    private let endCallButton =  EndCallButton.endCallButton()
+    private let handleView = AccessibilityActionView()
+    private let largePickUpButton = PickUpButton.bigPickUpButton()
+    private let largeHangUpButton = EndCallButton.bigEndCallButton()
+
+    private var establishedCallButtons: [IconLabelButton] {
+        return [flipCameraButton, cameraButton, microphoneButton, speakerButton, endCallButton]
+    }
+
+    var isIncomingCall: Bool = false {
+        didSet {
+            guard oldValue != isIncomingCall else { return }
+            topStackView.removeSubviews()
+            handleView.isHidden = isIncomingCall
+            handleView.accessibilityElementsHidden = isIncomingCall
+            if isIncomingCall  {
+                [microphoneButton, cameraButton, speakerButton].forEach(topStackView.addArrangedSubview)
+                addIncomingCallControllButtons()
+                verticalStackViewTopContraint.constant = 16.0
+            } else {
+                establishedCallButtons.forEach(topStackView.addArrangedSubview)
+                removeIncomingCallControllButtons()
+                verticalStackViewTopContraint.constant = 8.0
+            }
+            topStackView.distribution = isIncomingCall ? .equalSpacing : .fillEqually
+            setNeedsDisplay()
+        }
+    }
+
+    // MARK: - Setup
+
+    init() {
+        super.init(frame: .zero)
+
+        videoButtonDisabledTapRecognizer = UITapGestureRecognizer(target: self, action: #selector(performButtonAction))
+        setupViews()
+        createConstraints()
+    }
+
+    @available(*, unavailable) required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    private func setupViews() {
+        backgroundColor = SemanticColors.View.backgroundDefaultWhite
+        topStackView.distribution = .fillEqually
+        topStackView.spacing = 16
+        verticalStackView.alignment = .center
+        verticalStackView.spacing = 10
+        addSubview(verticalStackView)
+        establishedCallButtons.forEach(topStackView.addArrangedSubview)
+        handleView.layer.cornerRadius = 3.0
+        handleView.backgroundColor = SemanticColors.View.backgroundDragBarIndicator
+        [handleView, topStackView].forEach(verticalStackView.addArrangedSubview)
+        [flipCameraButton, cameraButton, microphoneButton, speakerButton, endCallButton, largeHangUpButton, largePickUpButton].forEach { $0.addTarget(self, action: #selector(performButtonAction), for: .touchUpInside) }
```

Comment:
suggestion to break up this long line 
```suggestion
        [
            flipCameraButton, 
            cameraButton, 
            microphoneButton,
            speakerButton, 
            endCallButton, 
            largeHangUpButton, 
            largePickUpButton
        ].forEach { 
            $0.addTarget(self, action: #selector(performButtonAction), for: .touchUpInside) 
        }
```